examples/server/utils.hpp

```diff
@@ -167,6 +167,34 @@ static T json_value(const json &body, const std::string &key, const T &default_v
         : default_value;
 }
 
+inline std::string format_llama2(std::vector<json> messages)
+{
+    std::ostringstream output;
+    bool is_inside_turn = false;
+
+    for (auto it = messages.begin(); it != messages.end(); ++it) {
+        if (!is_inside_turn) {
+            output << "<s>[INST] ";
```

Comment:
I suspect that later in the logic we might be incidentally adding some extra BOS tokens through `llama_tokenize(...)`. It's a bit difficult to follow the logic - probably need to debug this and add some more trace logs. If you feel like it, you might want to look into this

Reply:
Yeah you're right, when I tested with chat completion, `add_bos` is set to true, so I'll remove the `<s>` here

Reply:
The redundant `<s>` is removed in https://github.com/ggerganov/llama.cpp/pull/5425/commits/7efef47d2eb09160fc612c4edb811d261436e256

Validation for `--chat-template` is added in https://github.com/ggerganov/llama.cpp/pull/5425/commits/ebe30795390a4a5818d5b7c7bf2156dc2fd0312a

Please review and merge if it's OK for you. Thank you!

Reply:
Ok, waiting for @cebtenzzre's review